Fix initial github workflow

lefessan opened the pull request.

GitMensch commented:
As there are no other users and the commit is not too old, I'd suggest to place all those changes into the previous one and force-push the branch afterwards (so one instead of 2-N commits for the workflow).

Note: It would be nice to have both the PDF and the HTML versions generated.
Another note: you likely don't need `libtool libdb5.3-dev libxml2-dev libcjson-dev bison flex help2man gettext` in this workflow.

lefessan commented:
Yes, I will squash everything in the previous commit. I was forced to push that first commit, otherwise Github would not start testing it :-(

lefessan commented:
Note that I get the same error locally on my computer: `makeinfo` segfaults on gnucobpg, but not on other files. This is a known behavior ?

lefessan committed:
Fixes to get html generation to work

* Rename @anchoridx{Data} to @anchoridx{DateTypes} in section 13 (duplicate
  with the same index in section 7)
* Remove @validatemenus not available after version 6.8 of texinfo
* Rename section "cobc - The GnuCOBOL Compiler - Runtime options" to
  "cobc - The GnuCOBOL Compiler"
* Add @iftex around menu of section 7.8, as it makes makeinfo segfault and
  is not useful for HTML
* Small additional fixes for info generation to work

lefessan commented:
@vbcoen Wondering if you want to review the changes to .texi files in this PR ?

GitMensch commented on guide/gnucobpg.texi:
```diff
@@ -90,9 +90,9 @@ Permission is granted to copy and distribute translations of this manual into an
-@end ifnottex
+@comment @ifnottex
+@comment @validatemenus off
+@comment @end ifnottex
```
this can be dropped, no need for a comment

lefessan replied:
done

GitMensch commented on guide/cbrese.tex:
trailing tabs are never useful, so should all get out
all places before (C/S) should have a tab (currently they have mostly either a space or multiple spaces).

lefessan replied:
> trailing tabs are never useful, so should all get out all places before (C/S) should have a tab (currently they have mostly either a space or multiple spaces).

Done. Yet, the meaning of C/S is documented nowhere.

GitMensch replied:
Thank you, that's context/sensitive, depending on how the result looks like this may be a table and in this case it could be a headline with a checkbox.. or something

GitMensch commented on guide/7.texi:
```diff
@@ -663,6 +663,7 @@ As part of the @code{TIMES} clause of a @statementref{PERFORM}.
 @comment *********************************************************************
 @page
 @newsection{7.8,GnuCOBOL Statements}
+@iftex
```
why should this be removed for non-tex - and what is that?

lefessan replied:
It makes `makeinfo --html` segfault, and I have no idea why... Given that these lines are not useful in html mode, it was the simplest solution...

GitMensch replied:
please add a `@comment` before that with an explanation and ideally reference the upstream bug report, too

lefessan replied:
done

GitMensch reviewed (changes requested):
this PR currently mixes changes that are:
* for the mirror only (should be here, actually stashed to the previous commit)
* are reasonable tex internal changes  -> should get into upstream directly
* have some changes at least I don't understand

Note that "cobc - The GnuCOBOL Compiler - Runtime options" doesn't make any sense - either that refers to dialect or compile options or this is not related to cobc in the first place

GitMensch commented:
> Note that I get the same error locally on my computer: `makeinfo` segfaults on gnucobpg, but not on other files. This is a known behavior ?

No. Also "some time ago" it definitely did work.
Do you still have the segfault after the changes?

If there is a segfault in makeinfo, then there's definitely a bug there; please send reproducible steps with a link to the files needed to the appropriate bug tracker.
Ideally you can download its debug package and start the process under gdb, to include a `backtrace` when that happens to the bug report.

vbcoen commented:
I cannot see where these changes are stored and available to preview, at least with the changes to the indexing and menu etc.

Are they available ?

lefessan commented:
> I cannot see where these changes are stored and available to preview, at least with the changes to the indexing and menu etc.
> 
> Are they available ?

I moved the changes to the indexing/menu in another PR, this one just fixes html generation on the Github Actions.

For any PR, if you want to see the generated files, you can go in the corresponding checks -> Ubuntu Workflow -> Summay -> Artefacts, there should be one zip for all pdfs, and another one for html pages.

lefessan commented:
> Note that "cobc - The GnuCOBOL Compiler - Runtime options" doesn't make any sense - either that refers to dialect or compile options or this is not related to cobc in the first place

I just fixed the reference. That's the original title of the corresponding section. I understand it as "there are the options that the compiler accepts at runtime", by opposition with the options that can be used to build the compiler ("compile time option").

GitMensch commented:
Please adjust this - the only "real" compile time options are which libraries are used (the others don't make much difference for the use of cobc).

GitMensch commented on guide/8.texi:
```diff
@@ -1506,7 +1506,7 @@ When the @code{LOCALE} keyword is not specified, the returning value is negative
 @comment *********************************************************************
 @page
-@newsubsection{8.1.70B,NUMVAL-C-V2}
+@newsubsection{8.1.70B,NUMVAL-C-2}
```
Do you have an idea what this "B" is? Looks like two times the identical thing, no?
@vbcoen Did I miss something here? [if it is to be removed, then this should be done after the technical change which removes the renumbering problem, @lefessan could commit both the technical and "minor tweaks" change if you agree.

lefessan replied:
Indeed, it looks like it documents the same function twice, with a different explanation. Maybe somebody started improving the description by copying it temporarily, but never finished the job ?
